packages/contracts/contracts/optimistic-ethereum/OVM/accounts/OVM_ECDSAContractAccount.sol

```diff
@@ -66,66 +59,46 @@ contract OVM_ECDSAContractAccount is iOVM_ECDSAContractAccount {
             bytes memory
         )
     {
-        bool isEthSign = _signatureType == Lib_OVMCodec.EOASignatureType.ETH_SIGNED_MESSAGE;
+        // Attempt to decode the transaction.
+        Lib_EIP155Tx.EIP155Tx memory transaction = Lib_EIP155Tx.decode(
+            _encodedTransaction,
+            Lib_ExecutionManagerWrapper.ovmCHAINID()
```

Comment:
This can be handled with inline assembly instead of the wrapper if preferable.

Reply:
I originally had inline assembly but I think the wrapper is preferable.

Reply:
yeah, I am pretty surprised solidity doesn't expose chainID in a nicer way.